### DISCUSSION TITLE
Does it exist a method to compute the RGB to XYZ conversion matrix?

### DISCUSSION THREAD

**Discussion by stereomatchingkiss** [colour-science/colour#962] [General]
### Question

Do this library exist dictionary or method to compute the conversion matrix between rgb and xyz colorspace list on [this page](http://www.brucelindbloom.com/index.html?Eqn_RGB_XYZ_Matrix.html)? Example, 3.12.26 of readme.md hardcode matrix_XYZ_to_RGB value manually, do we have an api to get this matrix? Thanks

**Comment by thomasmansencal** [comment]
Hi @stereomatchingkiss,

Yes it can be done with the [colour.normalised_primary_matrix](https://colour.readthedocs.io/en/develop/generated/colour.normalised_primary_matrix.html) definition.

Cheers,

Thomas

**Reply by stereomatchingkiss** [reply]
Thanks, how could I found the primaries xy chromaticity coordinates?

**Reply by blue-j** [reply]
Thomas, thank you profoundly for sharing this incredible work. Deep bow!

I was wondering, reading this discussion, I have the impression the library doesn't have a conversion command?

Something like (pseudo code):

Lab = (15,117,-2)
XYZ = convert(Lab,XYZ)

?

Or in this case something like 

XYZ = convert(RGB,XYZ) -D50 -Bradford

Just want to be sure I haven't missed something.

**Reply by thomasmansencal** [reply]
> Thanks, how could I found the primaries xy chromaticity coordinates?

The primaries for the various RGB colourspaces we defined are available here: https://github.com/colour-science/colour/tree/develop/colour/models/rgb/datasets

> Thomas, thank you profoundly for sharing this incredible work. Deep bow!
> 
> I was wondering, reading this discussion, I have the impression the library doesn't have a conversion command?
> 
> Something like (pseudo code):
> 
> Lab = (15,117,-2) XYZ = convert(Lab,XYZ)
> 
> ?
> 
> Or in this case something like
> 
> XYZ = convert(RGB,XYZ) -D50 -Bradford
> 
> Just want to be sure I haven't missed something.

We have the `colour.convert` definition here that does what you are describing: https://colour.readthedocs.io/en/develop/generated/colour.convert.html#colour.convert
